language/move-prover/spec-lang/src/translate.rs

```diff
@@ -1423,7 +1440,7 @@ impl<'env, 'translator> ModuleTranslator<'env, 'translator> {
 impl<'env, 'translator> ModuleTranslator<'env, 'translator> {
     /// Definition analysis for move functions.
     /// If the function is pure, we translate its body.
-    fn def_ana_fun(&mut self, name: &PA::FunctionName, body: &EA::FunctionBody) {
+    fn def_ana_fun(&mut self, name: &PA::FunctionName, body: &EA::FunctionBody, fun_idx: usize) {
```

Comment:
The fun_idx parameter is confusing. Below you use SpecFunId::new(fun_idx), but at the caller side you use the definition index of the Move function. How is that possible? Wouldn't that clash with spec fun ids created from real spec funs? Whatever is going on here, please use more descriptive parameter names.

Reply:
Will use use more descriptive parameter names.

Because the `decl_ana` for move functions comes before spec functions, the `SpecDecl` for move functions will be pushed into `ModuleTranslator.spec_funs` before the `SpecDecl` for spec functions. Therefore, the definition index of the Move function is the same as the spec fun index for the spec version of that Move function, and it won't clash with real spec funs.